jodd-core/src/perf/java/jodd/util/Base64Benchmark.java

```diff
@@ -0,0 +1,119 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+package jodd.util;
+
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.UnsupportedEncodingException;
+
+
+/**
+ * Benchmark for encoding and decoding base64 - data.<br/>
+ * Following <tt>encode</tt>-methods will be compared:
+ * <ol>
+ *     <li>{@link java.util.Base64.Encoder#encodeToString(byte[])}</li>
+ *     <li>{@link org.apache.commons.codec.binary.Base64#encodeToString(byte[])}</li>
+ *     <li>{@link Base64#encodeToString(byte[])}</li>
+ * </ol>
+ *
+ * And following <tt>decode</tt>-methods will be compared:
+ * <ol>
+ *     <li>{@link java.util.Base64.Decoder#decode(String)}</li>
+ *     <li>{@link org.apache.commons.codec.binary.Base64#decode(String)}</li>
+ *     <li>{@link Base64#decode(String)}</li>
+ * </ol>
+ *
+ * <p>
+ * Run:
+ * <code>
+ * gw :jodd-core:perf -PBase64Benchmark
+ * </code>
+ * </p>
+ *
+ * results :
+ * <pre>
+ * Benchmark                                                        Mode  Cnt         Score        Error  Units
+ * Base64Benchmark.encode_Java_Base64                              thrpt   10   1775775,115 ± 107760,428  ops/s
+ * Base64Benchmark.encode_Apache_Base64                            thrpt   10    280928,427 ±  10380,474  ops/s
+ * Base64Benchmark.encode_Jodd_Base64                              thrpt   10   1641340,867 ±  39508,213  ops/s
+ *
+ * Benchmark                                                        Mode  Cnt         Score        Error  Units
+ * Base64Benchmark.decode_Java_Base64                              thrpt   10   1052911,630 ±  24643,940  ops/s
+ * Base64Benchmark.decode_Apache_Base64                            thrpt   10    254041,004 ±   3955,587  ops/s
+ * Base64Benchmark.decode_Jodd_Base64                              thrpt   10   1388376,060 ±  86304,922  ops/s
+ * </pre>
+ */
+@Fork(1)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+@State(Scope.Benchmark)
+public class Base64Benchmark {
+
+    private byte[] to_be_encoded ;
+    private String to_be_decoded ;
+
+	@Setup
+	public void prepare() throws UnsupportedEncodingException {
+		to_be_encoded = "Jodd is set of Java microframeworks, tools and utilities, under 1.7 MB. We believe in common sense to make things simple, but not simpler. Get things done! Make your ideas! Kickstart your startup! And enjoy the coding.".getBytes("UTF-8");
+		to_be_decoded = "Sm9kZCBpcyBzZXQgb2YgSmF2YSBtaWNyb2ZyYW1ld29ya3MsIHRvb2xzIGFuZCB1dGlsaXRpZXMsIHVuZGVyIDEuNyBNQi4gV2UgYmVsaWV2ZSBpbiBjb21tb24gc2Vuc2UgdG8gbWFrZSB0aGluZ3Mgc2ltcGxlLCBidXQgbm90IHNpbXBsZXIuIEdldCB0aGluZ3MgZG9uZSEgTWFrZSB5b3VyIGlkZWFzISBLaWNrc3RhcnQgeW91ciBzdGFydHVwISBBbmQgZW5qb3kgdGhlIGNvZGluZy4=";
+	}
+
+    // ----------------------------------------------------------------------- Java Base64
+    
+    @Benchmark
+    public void encode_Java_Base64(Blackhole blackhole) {
```

Comment:
For such simple test, you can simply return the computation result instead of consuming from a BlackHole.

```java
@Benchmark
public String encode_Java_Base64(Blackhole blackhole) {
  return java.util.Base64.getEncoder().encodeToString(to_be_encoded);
}
```

Reply:
true. Returned value is going to be consumed by a black hole. will refactor it now.

Reply:
Thank you for the hint, @slandelle !